lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -33,6 +33,21 @@
 
 TASK_LOG = logging.getLogger('edx.celery.task')
 
+ENROLLED_IN_COURSE = 'enrolled'
+
+NOT_ENROLLED_IN_COURSE = 'unenrolled'
+
+
+def _user_enrollment_status_mapping(course_id):
+    """
+    Returns a mapping of user ids to enrollment status in the given course.
+    """
+    return dict(CourseEnrollment.objects.values_list('user_id', 'is_active').filter(course_id=course_id))
```

Comment:
I don't know if this suggestion is any cleaner, but there are a few spots where you check a value from this dict and return either `ENROLLED_IN_COURSE` or `NOT_ENROLLED_IN_COURSE` based on that value. I don't think I see any other uses of this dict. 
So this data structure could be a defaultdict with a default value of `NOT_ENROLLED_IN_COURSE` and the user IDs mapped to `ENROLLED_IN_COURSE` or `NOT_ENROLLED_IN_COURSE` via the value of `is_active`. That way you could just get enum values directly from `_user_enrollment_status_mapping` instead of switching on the mapped value.
I think `_user_enrollment_status_mapping` could also be a lazy attribute.